subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/codecs/EnumCodec.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution.serialization.codecs
+
+import org.gradle.instantexecution.serialization.ReadContext
+import org.gradle.instantexecution.serialization.WriteContext
+
+
+object EnumCodec : EncodingProducer, Decoding {
+
+    override fun encodingForType(type: Class<*>): Encoding? =
+        EnumEncoding.takeIf { type.isEnum }
+
+    override suspend fun ReadContext.decode(): Any? {
+        val enumClass = readClass()
+        val enumOrdinal = readInt()
```

Comment:
⭕️ consider adapting/reusing `readEnum` here

Reply:
`writeEnum` and `readEnum` are different, they work when you have the actual enum type in the source, they are an optimization storing only the ordinal in that case. I would leave them as is. Did I miss something?